plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -259,6 +259,23 @@ protected function prepare_objects_query( $request ) {
 			}
 		}
 
+		if ( ! empty( $request['global_unique_id'] ) ) {
+			$global_unique_ids = explode( ',', $request['global_unique_id'] );
```

Comment:
If there are spaces around the `,` this will result in non-matches.

I suggest using `trim` to get rid of the whitespace...

```php
$global_unique_ids = array_map( 'trim', explode( ',', $request['global_unique_id'] ) )
```